src/services/cache-config/CacheConfigService.ts

```diff
@@ -315,10 +315,13 @@ export class CacheConfigService {
         this.createHelper(builder, address.concat("indexing", "INDEXING"), config.indexing);
         this.createHelper(builder, address.concat("compatibility", "COMPATIBILITY"), config.compatibility);
         this.createHelper(builder, address.concat("partition-handling", "PARTITION_HANDLING"), config["partition-handling"]);
-        this.createHelper(builder, address.concat("transaction", "TRANSACTION"), config.transaction);
+        if (this.isValidTXMode(builder, address, config)) {
+          this.createHelper(builder, address.concat("transaction", "TRANSACTION"), config.transaction);
+        } else {
+          deferred.reject("Cannot set TX mode to FULL_XA with ASYNC cache mode");
+        }
```

Comment:
Should we place this logic in some kind of `createTransactions` method, similar to `createCacheStore` etc, so that the logic remains self contained?

Also, do we not need to perform this check when updating a cache's config as well?